src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6721,11 +6650,6 @@ private ExpressionAst ExpressionRule(bool endNumberOnTernaryOpChars = false)
             // G      binary-expression   '?'   new-lines:opt   ternary-expression   new-lines:opt   ':'   new-lines:opt   ternary-expression
 
             // TODO: remove this if-block when making 'ternary operator' an official feature.
-            if (!ExperimentalFeature.IsEnabled("PSTernaryOperator"))
```

Comment:
Counterpoint here: if we want a backward compatible parser/tokenizer, we should keep some kind of static flags in so that we could eventually create an implementation where you can build instances of parsers targeting different versions or feature sets.

Reply:
Just preserving some element of the code in here will preserve enough information for us to unpick all this later.

Reply:
Would a comment be sufficient?

```csharp
// Added in PS7.0
```

Reply:
Comments won't really describe the way the logic branches, but if we really want to avoid different logic (or changing the conditions to `const bool`s) then we can muddle through with something designed to look and be machine parseable for a regex like `// ps7-expfeat:PSTernaryOperator` or something

Reply:
If we really want to pursue the `build instances of parsers targeting different versions or feature sets` goal, then we will have to guard the to-be-removed code in a compiler definition, like `#if PSV6 ... #endif`. I can see why we want this but it won't really help unless we test those guarded code on regular basis, and eventually this will make the code difficult to maintain.

Reply:
Perhaps we should revisit this later.